src/jit/codegenxarch.cpp

```diff
@@ -7890,7 +7890,89 @@ CodeGen::getBaseVarForPutArgStk(GenTreePtr treeNode)
     {
         // The first baseVarNum is guaranteed to be the first incoming arg of the method being compiled.
         // See lvaInitTypeRef() for the order in which lvaTable entries are initialized.
+        // For Windows the value is always 0 (there are always slots in the caller's frame for the first 4 args to a callee).
+        // For System V the first stack arg should be calculated.
         baseVarNum = 0;
+
+#ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
+        CORINFO_ARG_LIST_HANDLE argLst = compiler->info.compMethodInfo->args.args;
+        const unsigned argSigLen = compiler->info.compMethodInfo->args.numArgs;
```

Comment:
Does this arg list contain the following implicit or explicit args that JIT adds to a call node

Thisptr, generic context, Pinvoke cookie, vararg cookie and non-standard args added for indirect/VSD calls (see HasNonStandardArgs())

(I know vararg calls are not supported by coreclr, just mentioned above for completeness)